app/javascript/pages/DirectoryPage.tsx

```diff
@@ -12,19 +13,91 @@ interface DirectoryProps {
   assetPaths: unknown
 }
 
-const DirectoryPage = (props: DirectoryProps) => {
+interface ListingsGroup {
+  open: Listing[]
+  upcoming: Listing[]
+  results: Listing[]
+}
+
+const openListingsView = (listings) =>
+  listings.length > 0 ? (
+    <ListingsList listings={listings} />
```

Comment:
this is really interesting. I didn't realize that this was all tied into a "listings group" component. I sort of assumed that we'd pass in a list of listing directory objects with Image cards and tables, etc. 

Do you have a preference for what a better interface might be? I think we can probably talk to Sean tomorrow when Anthony and I meet with him at 2pm. You are also welcome to join if you'll still be online (I realize it's like 10pm your time).

Reply:
Regarding it being just one component I think we need to go the way you assumed it to be. Now we have little control over customization. If we make components less dependent and use flat structure passing singular params rather than whole listing would be a viable way to go. And in that case we are not going to need adapter.